fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/MetricUnionAvailabilitySpec.groovy

```diff
@@ -0,0 +1,233 @@
+                (metricColumn1): Collections.emptyList()
+        ]
+        availability2.getAllAvailableIntervals() >> [
+                (metricColumn2): Collections.emptyList()
+        ]
+
+        MetricColumn tableColumn1 = new MetricColumn("shouldNotBeReturned1")
+        MetricColumn tableColumn2 = new MetricColumn("shouldNotBeReturned2")
+
+        schema1.getColumns(_) >> Sets.newHashSet(tableColumn1)
+        schema2.getColumns(_) >> Sets.newHashSet(tableColumn2)
+
+        physicalTable1.getSchema() >> schema1
+        physicalTable2.getSchema() >> schema2
+
+        metricUnionAvailability = new MetricUnionAvailability(physicalTables, [metricColumn1, metricColumn2] as Set)
+
+        when:
+        Map<Availability, Set<MetricColumn>> availabilitiesToAvailableColumns = metricUnionAvailability.availabilitiesToAvailableColumns
+
+        then:
+        availabilitiesToAvailableColumns.size() == 2
+
+        availabilitiesToAvailableColumns.containsKey(availability1)
+        Set<MetricColumn> availableColumns1 = availabilitiesToAvailableColumns.get(availability1)
+        availableColumns1.contains(metricColumn1)
+        !availableColumns1.contains(tableColumn1)
+        !availableColumns1.contains(tableColumn2)
+
+        availabilitiesToAvailableColumns.containsKey(availability2)
+        Set<MetricColumn> availableColumns2 = availabilitiesToAvailableColumns.get(availability2)
+        availableColumns2.contains(metricColumn2)
+        !availableColumns2.contains(tableColumn1)
+        !availableColumns2.contains(tableColumn2)
+    }
+
+    @Unroll
+    def "checkDuplicateValue returns duplicates of metric column names from 2 physical tables in the case of #caseDescription"() {
+        when:
+        Map<MetricColumn, Set<Availability>> actual = MetricUnionAvailability.getDuplicateValue(
+                [
+                        (availability1): metricColumnNames1.collect{it -> new MetricColumn(it)} as Set,
+                        (availability2): metricColumnNames2.collect{it -> new MetricColumn(it)} as Set
+                ]
+        )
+
+        then:
+        actual.size() == expected.size()
+        for (Map.Entry<String, Integer> entry : expected.entrySet()) {
+            MetricColumn duplicate = new MetricColumn(entry.key)
+            actual.containsKey(duplicate)
+            actual.get(duplicate).size() == entry.value
+        }
+
+        where:
+        metricColumnNames1     | metricColumnNames2 | expected       | caseDescription
+        []                     | []                 | [:]            | 'both physical tables having empty metric column names'
+        ['metric1']            | []                 | [:]            | 'one empty table and one non-empty table'
+        ['metric1']            | ['metric1']        | ['metric1': 2] | 'both physical tables having the same metric column names'
+        ['metric1', 'metric2'] | ['metric1']        | ['metric1': 2] | 'two tables having intersections'
```

Comment:
It would be nice to add a case where there is more than 1 metric that has a duplicate, just to check it.